Lib/inspect.py

```diff
@@ -1932,7 +1932,9 @@ def _signature_get_partial(wrapped_sig, partial, extra_args=()):
             if param.kind is _POSITIONAL_ONLY:
                 # If positional-only parameter is bound by partial,
                 # it effectively disappears from the signature
-                new_params.pop(param_name)
+                # However, if it is a Placeholder it is not removed
+                if arg_value is not functools.Placeholder:
+                    new_params.pop(param_name)
```

Comment:
This is not covered by tests.

Reply:
Add also a test where Placeholder replaces an optional positional-only parameter.